### PR TITLE
Subplot groups

### PR DESCRIPTION
**Pull request by dcbr**
### Overview

I implemented this for one of my projects and thought it might be useful for others to use as well. This allows individual renderers to span multiple rows and columns of the layout grid, similar to what can be done with subplots in matplotlib and Matlab.

### Details

- Adds a 'groups' argument to BasePlotter which should be a list or tuple defining all groups for this plotter. Each group can be defined through a slice or sequence of row and column ids (see multi-window.py example). This allows for more complicated UI designs that could previously not be achieved through the 'shape' argument as far as I know.

### CODE REVIEW

**Comment by banesullivan** [comment]
Now, this looks cool! Thanks for the hard work @dcbr!

sphx_glr_multi-window_004

It would be great to write up a bit more in the example for how to use this

**Comment by dcbr** [comment]
Ok, glad you like it! I can write some more information on the usage later, the rest I leave to you then :)

**Comment by banesullivan** [comment]
Also an FYI to myself or whoever comes back to this, some changes in here are relevant to #626 and may need to be reworked

**Comment by dcbr** [comment]
I generalized it slightly, allowing to specify the 'weight' of each column and row in the grid. Let me know what you think! I also included this in the multi-window example.

**Comment by dcbr** [comment]
Any update on this? Is there something I should add or change?
I see the documentation build failed, however I don't see the problem in the the example I changed. If I run it without building the documentation everything runs fine.

**Comment by akaszynski** [comment]
Let me take a look at this on my end.

**Comment by akaszynski** [comment]
There's some fixes that need to be applied to master first.  I'll fix that and then get back to this.

**Comment by akaszynski** [comment]
@dcbr, could you please merge to `pyvista`'s master?

**Comment by dcbr** [comment]
I'm not very familiar with pull requests, what exactly should I do? Currently it says my branch is 5 commits ahead of pyvista:master (after your latest commit).

**Comment by akaszynski** [comment]
What you need to do is to update your repo with the latest pyvista master.  See below for directions for how to do this:

https://digitaldrummerj.me/git-sync-fork-to-master/

I’ve dealt with the merge conflicts, so you should be able to merge cleanly.

**Comment by dcbr** [comment]
Thanks for the link, learned something new :)

**Comment by akaszynski** [comment]
I'm seeing a strange bug that only occurs on Azure CI that I'm patching in #771.  Once that's merged, could you merge with master one last time?  Once that's done we should be good to go for supporting this new feature.

**Comment by akaszynski** [comment]
#771 has been merged.  Can you merge with master one more time and test it out?  It should be good to go.

**Comment by dcbr** [comment]
Thanks for figuring out what caused this documentation fail!

**Comment by akaszynski** [comment]
I see the issue.  Let me patch this quickly.

File: examples/02-plot/multi-window.py [hunk 1]
```diff
@@ -90,3 +90,74 @@
...
+groups = [
+    (0,np.s_[:]),      # First group spans over all columns of the first row (0)
+    ([1,3],0),         # Second group spans over row 1-3 of the first column (0)
+    (np.s_[2:],[1,2]), # Third group spans over rows 2-4 and columns 1-2
+    (slice(1,-1),3)    # Fourth group spans over rows 1-3 of the last column (3)
+]
+
+plotter = pv.Plotter(shape=shape,row_weights=row_weights,col_weights=col_weights,groups=groups)
+
+###############################################################################
```
Comment (akaszynski) [inline examples/02-plot/multi-window.py:116]:
```suggestion
```

**Comment by akaszynski** [comment]
Found the problem when using this locally.  Having the plotter initialization and `add_mesh` separated from one another with:
```python
###############################################################################
```
Causes the instance of `Plotter` to be cleaned.  Once it's cleaned, we see some strange behavior like the plotter being unable to find its renderer.  This issue is tough to duplicate within just a script as the script doesn't collect anything between the `###########` breaks.  I think there's some behavior internal to `sphinx_gallery` that causes figures to be collected at the end of a block, and therefore we can't end a block until `plotter.show()`.

**Pull request merged**